### PR TITLE
New http api to nomos-node integration

### PR DESCRIPTION
**Pull request by al8n**
Integrate new http api crate to nomos-node.

### CODE REVIEW

File: nodes/nomos-node-api/src/lib.rs [hunk 1]
```diff
@@ -21,16 +22,17 @@ pub trait Backend {
...
         Self: Sized;
 
-    async fn serve(self) -> Result<(), Self::Error>;
+    async fn serve(self, handle: OverwatchHandle) -> Result<(), Self::Error>;
 }
 
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
 pub struct ApiServiceSettings<S> {
     pub backend_settings: S,
```
Comment (bacv) [inline nodes/nomos-node-api/src/lib.rs:30]: `backend` would be more consistent with what we have in other service settings, but if we go with `backend_settings`, then the `node/nomos-node/config.yaml` needs to be updated accordingly.

File: nodes/nomos-node-api/src/http/backend/axum.rs [hunk 2]
```diff
@@ -273,15 +311,18 @@ where
         (status = 500, description = "Internal server error", body = String),
     )
 )]
-async fn add_cert(State(store): State<OverwatchHandle>, Json(cert): Json<Certificate>) -> Response {
+async fn add_cert(
+    State(handle): State<OverwatchHandle>,
+    Json(cert): Json<Certificate>,
```
Comment (bacv) [inline nodes/nomos-node-api/src/http/backend/axum.rs:316]: This handle should expect Cert that is serialized with `wire::serialize`. At least `mempool-da/add` had it in such way. 
Expecting bytes and then deserializing it with `wire::deserialize::<Certificate>(&payload).unwrap()` works, but I think this would ultimately need it's own extractor.

Comment (al8n) [reply nodes/nomos-node-api/src/http/backend/axum.rs:316]: Ah, thanks! I think this should be the reason why the CI continues failing.

File: tests/Cargo.toml [hunk 3]
```diff
@@ -23,7 +23,7 @@ mixnet-topology = { path = "../mixnet/topology" }
 # Using older versions, since `mixnet-*` crates depend on `rand` v0.7.3.
 rand = "0.7.3"
 once_cell = "1"
-secp256k1 = { version = "0.26", features = ["rand"] }
+secp256k1 = { version = "0.28", features = ["rand"] }
```
Comment (danielSanchezQ) [inline tests/Cargo.toml:26]: Why do we update this here? Are we using something not available in other versions? Otherwise it would be better if it has its own pr.

File: nodes/nomos-node-api/src/lib.rs [hunk 4]
```diff
@@ -53,13 +55,16 @@ where
...
     }
 
     /// Service main loop
     async fn run(mut self) -> Result<(), DynError> {
         let endpoint = B::new(self.settings.backend_settings).await?;
-        endpoint.serve().await?;
+        endpoint.serve(self.handle).await?;
         Ok(())
     }
 }
```
Comment (danielSanchezQ) [inline nodes/nomos-node-api/src/lib.rs:70]: This probably needs to be updated. There are things to do here. Usually you would want the service to hold the whole service state.  Lifecycle is ready as well. **Not to be done now**, but lets open an issue to handle the service here. I can probably implement that myself.

File: nodes/nomos-node-api/src/http/backend/axum.rs [hunk 5]
```diff
@@ -18,10 +23,23 @@ use crate::{
...
+        default_value_t = std::net::SocketAddr::new(
+            std::net::IpAddr::V4(std::net::Ipv4Addr::new(127, 0, 0, 1)),
+            8080,
+        ),
+        env = "HTTP_BIND_ADDRESS"
+    ))]
+    pub address: std::net::SocketAddr,
+    /// Allowed origins for this server deployment requests.
+    #[cfg_attr(feature = "clap", arg(long = "http-cors-origin"))]
+    pub cors_origins: Vec<String>,
```
Comment (danielSanchezQ) [inline nodes/nomos-node-api/src/http/backend/axum.rs:42]: Why do we have clap here? Are we using it as standalone somewhere?

Comment (al8n) [reply nodes/nomos-node-api/src/http/backend/axum.rs:42]: In the old HTTP service, we have clap, so I just keep them in the new HTTP service. I have not seen we use it standalone.

File: nodes/nomos-node-api/tests/todo.rs [hunk 6]
```diff
@@ -71,7 +74,7 @@ impl Backend for WebServer {
         Ok(Self { addr: settings })
     }
 
-    async fn serve(self) -> Result<(), Self::Error> {
+    async fn serve(self, _handle: OverwatchHandle) -> Result<(), Self::Error> {
```
Comment (youngjoon-lee) [inline nodes/nomos-node-api/tests/todo.rs:77]: Maybe I'm missing something from previous PRs, but may I ask why this handle is not used?

Comment (danielSanchezQ) [reply nodes/nomos-node-api/tests/todo.rs:77]: Indeed, I think it is because is feed to the settings. @al8n ? I think in this case it should not be part of the settings itself. But should be expected to be received later on just before beeing served by the Backend. Unless there is a strong reason we can not.

Comment (al8n) [reply nodes/nomos-node-api/tests/todo.rs:77]: > Maybe I'm missing something from previous PRs, but may I ask why this handle is not used?

This handle is used to get a `Relay` for different services, in this test case, we only have one service (self), so this handle is not used.

File: nodes/nomos-node-api/src/lib.rs (hunk 4 above)
Comment (al8n) [reply nodes/nomos-node-api/src/lib.rs:70]: Created at https://github.com/logos-co/nomos-node/issues/500.

File: nodes/nomos-node-api/src/http/backend/axum.rs (hunk 5 above)
Comment (danielSanchezQ) [reply nodes/nomos-node-api/src/http/backend/axum.rs:42]: I think we should remove it if we do not use it.

File: nodes/nomos-node-api/tests/todo.rs (hunk 6 above)
Comment (youngjoon-lee) [reply nodes/nomos-node-api/tests/todo.rs:77]: Ah, I"m sorry. I didn't realize that this is a test :)

File: tests/Cargo.toml (hunk 3 above)
Comment (al8n) [reply tests/Cargo.toml]: This should be updated by vscode plugin. I will change it back.

File: nodes/nomos-node-api/src/http/backend/axum.rs (hunk 2 above)
Comment (al8n) [reply nodes/nomos-node-api/src/http/backend/axum.rs:306]: > tbh, if all other API are json we should be consistent and use json here as well unless there's a specific reason not to do so

Yes, make sense, I will try to make the CI works by using JSON args.

File: nodes/nomos-node/src/main.rs [hunk 7]
```diff
@@ -4,19 +4,16 @@ use nomos_node::{
...
 use color_eyre::eyre::{eyre, Result};
 use nomos_core::{
     da::{blob, certificate},
     tx::Transaction,
 };
-use nomos_http::bridge::{HttpBridge, HttpBridgeSettings};
-use nomos_mempool::network::adapters::libp2p::{Libp2pAdapter, Settings as AdapterSettings};
-use nomos_network::backends::libp2p::Libp2p;
+
+use nomos_mempool::network::adapters::libp2p::Settings as AdapterSettings;
```
Comment (danielSanchezQ) [inline nodes/nomos-node/src/main.rs:14]: Why is this change here? This is related to mempool right?

**Review by danielSanchezQ — APPROVED** [review]
Looking good! and CI is 🟢 !
Just left a tiny comment.

File: nodes/nomos-node/src/main.rs (hunk 7 above)
Comment (al8n) [reply nodes/nomos-node/src/main.rs:14]: Yes, we need this setting to construct a mempool service.

Comment (danielSanchezQ) [reply nodes/nomos-node/src/main.rs:14]: But why the adapter and the backend is gone now?

Comment (al8n) [reply nodes/nomos-node/src/main.rs:14]: In the previous, we had type aliases for creating bridges.
